webhooks-extension/pkg/endpoints/sink.go

```diff
@@ -245,6 +248,98 @@ func createPipelineRunFromWebhookData(buildInformation BuildInformation, r Resou
 	logging.Log.Debugf("PipelineRun created: %+v.", pipelineRun)
 }
 
+// This creates TaskRun for monitoring the main PipelineRun and reporting the result to the github
+func createTaskRunFromWebhookData(buildInformation BuildInformation, r Resource) {
+	logging.Log.Debugf("In createTaskRunFromWebhookData, build information: %s.", buildInformation)
+
+	installNs := os.Getenv("INSTALLED_NAMESPACE")
+	if installNs == "" {
+		installNs = "default"
+	}
+
+	logging.Log.Debugf("Looking for the pipeline configmap in the install namespace %s.", installNs)
+
+	// get information from related githubsource instance
+	webhook, err := r.getGitHubWebhook(buildInformation.REPOURL, installNs)
+	if err != nil {
+		logging.Log.Errorf("error getting github webhook: %s.", err.Error())
+		return
+	}
+
+	taskTemplateName := webhook.PullTask
+	taskNs := webhook.Namespace
+	saName := webhook.ServiceAccount
+	accessTokenRef := webhook.AccessTokenRef
+	OnSuccessComment := webhook.OnSuccessComment
+	OnFailureComment := webhook.OnFailureComment
+
+	if saName == "" {
+		saName = "default"
+	}
+
+	if taskTemplateName == "" {
+		taskTemplateName = "monitor-result-task"
+	}
+
+	if OnSuccessComment == "" {
+		OnSuccessComment = "OK"
+	}
+
+	if OnFailureComment == "" {
+		OnFailureComment = "ERROR"
+	}
+
+	logging.Log.Debugf("Build information: %+v.", buildInformation)
+
+	// Assumes you've already applied the yml: so the task definition must exist upfront.
+	startTime := buildInformation.TIMESTAMP
+	generatedPipelineRunName := fmt.Sprintf("%s-%s", webhook.Name, startTime)
+	generatedTaskRunName := generatedPipelineRunName
+
+	// Unique names are required so timestamp them.
+	pullRequestResourceName := fmt.Sprintf("%s-pull-request-%s", webhook.Name, startTime)
```

Comment:
webhook name might not end up being unique - so we might need an alternative uniqueness.  I'm wondering if you can not set Name but set GenerateName: "foo-"  ........... would mean having to change definePipelineResource(?) to allow you to choose whether you want a generated name or a specified name.

In the above, when using GenerateName: "foo-" rather than Name under ObjectMeta ..... you would get a generated unique name with foo- as a prefix.  So perhaps you could use webhook.Name- as the prefix and let kube take care of generating the unique suffix

Reply:
The way to generate the taskrun and the pullrequest resource names is the same way to generate the pipelinerun and the other resurce names before this PR.  I would like to address the unique name problem in a separate issue for all resources.
I created an issue to address the name uniqueness issue.